README.md

```diff
@@ -0,0 +1,41 @@
+# Grist-Pack: an all-in-one packaging of Grist for cloud providers
+
+This is a [Packer](https://developer.hashicorp.com/packer/docs/intro) build of Grist that generates virtual machine images for cloud providers. Currently Amazon Web Services and DigitalOcean are supported. The generated images will have a `grist` user that is mostly setup to run Grist via [Docker Compose](https://docs.docker.com/compose/).
+
+
+## Building the virtual machine images
+
+Follow these steps to build AMIs for AWS and Snapshots for DigitalOcean (DO).
+
+1. [Install Packer](https://developer.hashicorp.com/packer/install).
+2. clone this repository: `git clone https://github.com/gristlabs/grist-pack.git`
+3. Initialise packer: `cd grist-pack && packer init .`
+
+You now need to obtain credentials for AWS and DO.
+
+1. For AWS, you need to obtain an access key with its corresponding secret.
+2. For DO, you will need a token.
+
+Create a file called `grist.auto.pkrvars.hcl` and put those secrets into it. For example,
+
+```sh
+echo '
+aws_access_key = "XXXXXXXXXXXXXXXXXXXX"
+aws_secret_key = "YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY"
+do_token = "dop_v1_zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+' > grist.auto.pkrvars.hcl
+```
+
+You are now ready to build the images:
+
+```sh
+packer build .
+```
+
+This should take a few minutes and generate in parallel two images, named `grist-marketplace-${timestamp}`. If it completes successfully, it will write the resulting IDs of the image to a `manifest.json` file in the local directory. You can now go into your AWS console to inspect the resulting AMI or to your DigitalOcean dashboard and see the new snapshot. You may further launch virtual machines as normal via the usual means for each cloud provider.
+
+## Updating the images
+
+The Packer configuration can be found in the `grist.pkr.hcl` file. Two variables are defined there for choosing the base image for generating the AMI or snapshot: `aws_image_filter` and `do_image` for AWS and DigitalOcean respectively. These variables can be overriden [via the usual method for Packer variables](https://developer.hashicorp.com/packer/docs/templates/hcl_templates/variables#assigning-values-to-input-variables) in order to pick different base images to build from.
+
+For DigitalOcean, note that `scripts/digitalocean-img-check`, a linting script, has a hardcoded list of allowed base images. This script should be updated from its upstream source whenever DigitalOcean updates the allowed list.
```

Comment:
This README was enough for me to get images built, and to run an instance from them and start poking around (just tested on aws so far). So that's good!

Reply:
My experience with the AMI. I run an image from it, leaving everything at defaults since I'm coming from the README and haven't noticed any more direction yet. Instance comes up fine. I connect. At a terminal, type `ls`, don't see anything helpful. Do `find / -iname "grist*"` and fine `/home/grist`. Wonder if I should have connected as user `grist`? Go to that directory, see a README. Start following instructions. Run the secrets script, fine. Run `docker compose up` it looks pretty good but there's an authelia error and things stop (error about grist.local maybe). Kind of expected I'd need to do some configuration. Probably should read more of README now, that's as far as I've gotten.

Pretty good!

Reply:
Ok, something that works is to log in as ubuntu and then do `sudo su grist` and `cd`.
Maybe the README needs a small AWS section.

Reply:
Tried in Digital Ocean. Logging in as first thing I tried (root) worked. And following README to run Grist looked healthier in console. Visiting https://<public ip address> in a brain-dead way gave the `404 page not found` I associate in my mind with traefik. Encouraging, putting myself in the place of someone who doesn't know Grist, this is already not bad, I'm ready to start reading config documentation now.

Reply:
(Really appreciate that image includes at least nano and tmux)

Reply:
Ok, with a public domain in hand, I went to edit .env to taste. Did my best, ran `docker compose up`, some non-obvious problem. Ran `bin/generate-secure-secrets` again just in case. New problem. Eventually realize that .env was reset. Flailing. Need more instructions.

Reply:
So, another rough edge is that incomplete previous runs leave files with root perms under `persist`.

Try:

```sh
sudo rm -r persist config/secrets
GRIST_DOMAIN=your.public.domain ./bin/generate-secure-secrets
nano ~/.env  ## Fix the domain and the default email
docker compose down && docker compose up
```